### DISCUSSION TITLE
Same command, different output.

### DISCUSSION THREAD

**Discussion by mateosss** [ImageMagick/ImageMagick#2216] [Help] [answered]
Hello! I have Lubuntu 18.04 which has `ImageMagick 6.9.7-4 Q16` in it and a virtual machine with Lubuntu 20.04 with `ImageMagick 6.9.10-23 Q16`.
I ran `convert ubuntu.svg ubuntu.png` in both and their outputs differ but I need both to look like the `6.9.7-4` version.
Any ideas as to what flags could I use? I guess it may have something to do with the transparencies but not much more ideas than that.

**6.9.10-23 output png**
![6 9 10-23](https://user-images.githubusercontent.com/5622899/86079837-d23f8f00-ba67-11ea-841a-24175855fcf2.png)

**6.9.7-4 output png**
![6 9 7-4](https://user-images.githubusercontent.com/5622899/86079839-d370bc00-ba67-11ea-908e-6fcdbff295c2.png)

<details>
<summary>Original ubuntu.svg file (sorry can't directly upload it)</summary>

```svg

<?xml version="1.0" encoding="UTF-8"?><!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd"><svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" version="1.1" id="mdi-ubuntu" width="24" height="24" viewBox="0 0 24 24"><path d="M22,12A10,10 0 0,1 12,22A10,10 0 0,1 2,12A10,10 0 0,1 12,2A10,10 0 0,1 22,12M14.34,7.74C14.92,8.07 15.65,7.87 16,7.3C16.31,6.73 16.12,6 15.54,5.66C14.97,5.33 14.23,5.5 13.9,6.1C13.57,6.67 13.77,7.41 14.34,7.74M11.88,15.5C11.35,15.5 10.85,15.39 10.41,15.18L9.57,16.68C10.27,17 11.05,17.22 11.88,17.22C12.37,17.22 12.83,17.15 13.28,17.03C13.36,16.54 13.64,16.1 14.1,15.84C14.56,15.57 15.08,15.55 15.54,15.72C16.43,14.85 17,13.66 17.09,12.33L15.38,12.31C15.22,14.1 13.72,15.5 11.88,15.5M11.88,8.5C13.72,8.5 15.22,9.89 15.38,11.69L17.09,11.66C17,10.34 16.43,9.15 15.54,8.28C15.08,8.45 14.55,8.42 14.1,8.16C13.64,7.9 13.36,7.45 13.28,6.97C12.83,6.85 12.37,6.78 11.88,6.78C11.05,6.78 10.27,6.97 9.57,7.32L10.41,8.82C10.85,8.61 11.35,8.5 11.88,8.5M8.37,12C8.37,10.81 8.96,9.76 9.86,9.13L9,7.65C7.94,8.36 7.15,9.43 6.83,10.69C7.21,11 7.45,11.47 7.45,12C7.45,12.53 7.21,13 6.83,13.31C7.15,14.56 7.94,15.64 9,16.34L9.86,14.87C8.96,14.24 8.37,13.19 8.37,12M14.34,16.26C13.77,16.59 13.57,17.32 13.9,17.9C14.23,18.47 14.97,18.67 15.54,18.34C16.12,18 16.31,17.27 16,16.7C15.65,16.12 14.92,15.93 14.34,16.26M5.76,10.8C5.1,10.8 4.56,11.34 4.56,12C4.56,12.66 5.1,13.2 5.76,13.2C6.43,13.2 6.96,12.66 6.96,12C6.96,11.34 6.43,10.8 5.76,10.8Z" /></svg>

```
</details>

Thanks in advance!

**Comment by fmw42** [comment, marked as answer]
The issue is likely different software for rendering the SVG. ImageMagick can use one of 3 different tools: its own internal MSVG/XML, RSVG delegate or Inkscape. Inkscape is the best, followed by RSVG and lastly the ImageMagick MSVG/XML. Please check which one it is and what version. Be sure to use the same on your two computers.  You can check by adding -verbose to your convert command

`convert -verbose ubuntu.svg output.png`

You can tell MSVG/XML or RSVG by

`convert -list format`

and looking at the end of the line for SVG. That will give version numbers, but that will not identify Inkscape.  

Why can you not post your SVG file?  You can zip it and upload the zip file.

**Reply by mateosss** [reply]
Thank you! that was it, I installed inkscape on Lubuntu 20.04 and it gave the same output png.

However the output of `convert -verbose` said `inkscape` in both machines even though I did not have inkscape installed in 20.04.

**Lubuntu 20.04 Before Installing Inkscape**
```none
matter@MATTER:~/Desktop$ convert -list format | grep SVG
     MSVG  SVG       rw+   ImageMagick's own SVG internal renderer
      SVG  SVG       rw+   Scalable Vector Graphics (XML 2.9.10)
     SVGZ  SVG       rw+   Compressed Scalable Vector Graphics (XML 2.9.10)
matter@MATTER:~/Desktop$ convert -verbose ubuntu.svg ubuntu.png
'inkscape' '/tmp/magick-2152NpDD1hJUiVXW' --export-png='/tmp/magick-2152K3RCpuyCW8mV' --export-dpi='96,96' --export-background='rgb(100%,100%,100%)' --export-background-opacity='1' > '/tmp/magick-2152TJKm43j65QYT' 2>&1
mvg:/tmp/magick-2152YuCy9VEzq2oU=>/tmp/magick-2152YuCy9VEzq2oU MVG 24x24 24x24+0+0 16-bit sRGB 1500B 0.000u 0:00.000
ubuntu.svg MVG 24x24 24x24+0+0 16-bit sRGB 1500B 0.000u 0:00.000
ubuntu.svg=>ubuntu.png MVG 24x24 24x24+0+0 16-bit sRGB 35c 518B 0.010u 0:00.000
```

**Lubuntu 18.04 With Inkscape**

```none
mateo@BANER:~/Desktop$ convert -list format | grep SVG
     MSVG  SVG       rw+   ImageMagick's own SVG internal renderer
      SVG  SVG       rw+   Scalable Vector Graphics (XML 2.9.4)
     SVGZ  SVG       rw+   Compressed Scalable Vector Graphics (XML 2.9.4)
mateo@BANER:~/Desktop$ convert -verbose ubuntu.svg ubuntu.png
'inkscape' '/tmp/magick-48122VCeGtDebxxo' --export-png='/tmp/magick-4812AXZbScNkFLmq' --export-dpi='90,90' --export-background='rgb(100%,100%,100%)' --export-background-opacity='1' > '/tmp/magick-4812vBv6hkXUo0bs' 2>&1
/tmp/magick-4812AXZbScNkFLmq PNG 23x23 23x23+0+0 8-bit sRGB 722B 0.000u 0:00.000
ubuntu.svg SVG 23x23 23x23+0+0 8-bit sRGB 722B 0.000u 0:00.000
ubuntu.svg=>ubuntu.png SVG 23x23 23x23+0+0 8-bit sRGB 83c 605B 0.000u 0:00.000

```

In any case, thanks!

**Reply by fmw42** [reply]
In the first case, it looks for Inkscape and cannot find it so it uses **mvg:** Look carefully at the differences in the verbose response. You will see **mvg:** only in the first one.

**Comment by snibgo** [comment]
```
Before Installing Inkscape
...
ubuntu.svg=>ubuntu.png MVG 24x24
```
"MVG" tells you that IM actually used its internal MSVG to rasterize the image.
